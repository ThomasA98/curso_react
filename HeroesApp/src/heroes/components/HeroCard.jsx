import { useMemo } from "react";
import { Link } from "react-router-dom";

export const HeroCard = ({
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters,
}) => {

    const heroUrl = useMemo(() => `/assets/heroes/${id}.jpg` );
    
    return (
        <div className="col">
            <div className="card">
                <div className="row">
                    <div className="col-4">
                        <img className="img-fluid rounded-start" src={ heroUrl } alt={ id } />
                    </div>

                    <div className="col-8">
                        <div className="card-body">
                            <h5 className="card-title" >{ superhero }</h5>

                            <p className="card-text">{ alter_ego }</p>
                            
                            {  characters === alter_ego || <p>{ characters }</p> }

                            <p className="card-text">
                                <small className="text-muted">{ first_appearance }</small>
                            </p>

                            <Link to={ `/hero/${id}` }>
                                mas...
                            </Link>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};