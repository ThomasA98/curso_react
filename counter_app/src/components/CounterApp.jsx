import PropTypes from 'prop-types';
import { useCounter } from '../hooks/useCounter';
import styled from 'styled-components';

export const CounterApp =  ({ value })  => {

    const [
        counterValue,
        {
            increment,
            incrementAt ,
            decrement,
            decrementAt,
            reset
        }
    ] = useCounter(value);

    return (
        <CounterAppStyles>
            <h1 className='counter-title'>Counter App</h1>
            <h2 className='counter-number'>{ counterValue }</h2>
            <div className="counter-buttons">
                <button className='counter-btn' type="button" onClick={ increment } >increment</button>
                <button className='counter-btn' type="button" onClick={ decrement } >decrement</button>
                <button className='counter-btn' type="button" onClick={ incrementAt(2) } >increment 2</button>
                <button className='counter-btn' type="button" onClick={ decrementAt(2) } >decrement 2</button>
                <button className='counter-btn' type="button" onClick={ reset } >reset</button>
            </div>
        </CounterAppStyles>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
    value: 0,
};

const CounterAppStyles = styled.section`

    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: center;
    align-content: center;

    & > .counter-title {
        font-size: 60px;
        color: #484;
        text-transform: uppercase;
        text-align: center;
        margin-bottom: 15px;
    }

    & .counter-number {
        text-align: center;
        font-size: 50px;
    }

    & .counter-btn {
        display: block;
        width: 200px;
        height: 40px;
        font-size: 20px;
        margin: 5px 0;
        color: white;
        background-color: #484;
        border: none;
        border-radius: 10px;
    }

    & .counter-buttons {
        margin: 0 auto;
    }

`;