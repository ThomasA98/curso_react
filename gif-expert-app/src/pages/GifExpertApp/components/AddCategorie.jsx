import PropTypes from 'prop-types';
import { useState } from "react";

export const AddCategorie = ({ onSubmitCategorie }) => {

	const [inputCategorie, setInputCategorie] = useState('');

	const onInputCategorieChange = ({ target: { value } }) => {
		setInputCategorie(value);
	};

	const OnSubmitInput = (event) => {
		event.preventDefault();

		if (inputCategorie.trim()) onSubmitCategorie(inputCategorie.trim());

		setInputCategorie("");
	};

	return (
		<form onSubmit={OnSubmitInput} aria-label="form">
			<input
				type="text"
				onChange={onInputCategorieChange}
				value={inputCategorie} />
		</form>
	);
};

AddCategorie.propTypes = {
	onSubmitCategorie: PropTypes.func.isRequired
};