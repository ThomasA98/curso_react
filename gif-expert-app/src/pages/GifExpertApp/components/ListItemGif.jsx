import PropTypes from 'prop-types';

export const ListItemGifs = ({ title, img_ref }) => {
    return (
        <div className="card" >
            <p>{ title }</p>
            <img src={ img_ref } alt={ title } />
        </div>
    );
};

ListItemGifs.propTypes = {
    title: PropTypes.string.isRequired,
    img_ref: PropTypes.string.isRequired
};