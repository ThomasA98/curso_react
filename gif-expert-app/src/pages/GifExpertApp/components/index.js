export * from './AddCategorie';
export * from './ListCategories';
export * from './ListItemGif';