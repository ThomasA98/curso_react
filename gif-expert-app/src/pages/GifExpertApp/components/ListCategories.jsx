import { useGifFetch } from "../hooks/useGifFetch";
import { ListItemGifs } from "./ListItemGif";
import PropTypes from 'prop-types';

export const ListCategories = ({ gifToSerch }) => {

    const { gifs, isLoading } = useGifFetch(gifToSerch);

    return (
        <div>
            { isLoading && <p>Loading...</p> }
            <div className="card-grid" >
            {
                gifs?.map(({ id, img_ref, title }) =>
                    <ListItemGifs
                    key={ id }
                    img_ref={ img_ref }
                    title={ title }
                    />)
            }
            </div>

        </div>
    );
};

ListCategories.propTypes = {
    gifToSerch: PropTypes.string.isRequired,
}