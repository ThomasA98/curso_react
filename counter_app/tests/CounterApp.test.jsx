import { render, screen, fireEvent } from "@testing-library/react";
import { CounterApp } from './components/CounterApp';

describe('Testeando el componente <CounterApp />', () => {

    let initialCounter = 100;

    test('debe hacer match con el snapshot', () => {
        
        let { container } = render(<CounterApp value={initialCounter} />);

        expect(container).toMatchSnapshot();

    });

    test('debe mostrar el valor inicial', () => {
        
        render(<CounterApp value={initialCounter} />);
    
        expect(screen.getAllByText(initialCounter)).toBeTruthy();
    });

    test('debe incrementar con el boton increment', () => {

        render(<CounterApp value={initialCounter} />);

        fireEvent.click(screen.getByText('increment'));

        expect(screen.getByText(initialCounter + 1)).toBeTruthy();

    });

    test('debe decrementar con el boton decrement', () => {

        render(<CounterApp value={initialCounter} />);

        fireEvent.click(screen.getByText('decrement'));

        expect(screen.getByText(initialCounter - 1)).toBeTruthy();

    });

    test('debe resetear al valor iniciala con el boton reset', () => {

        render(<CounterApp value={initialCounter} />);

        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(initialCounter.toString());

    });

});