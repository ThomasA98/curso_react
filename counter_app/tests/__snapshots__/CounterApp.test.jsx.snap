// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeando el componente <CounterApp /> debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      Counter App
    </h1>
    <h2
      data-testid="test-counter-value"
    >
      100
    </h2>
    <button>
      increment
    </button>
    <button>
      decrement
    </button>
    <button
      aria-label="btn-reset"
    >
      reset
    </button>
  </div>
</div>
`;
